### PR TITLE
fixed failed to load api specification error

### LINKED ISSUE #357: 'Failed to load api specification' error when visiting https://localhost:8000
**Describe the bug**
when you build zubhub on your localmachine and try visiting https://localhost:8000 on your browser, you get an internal server error with message "failed to load api specification". This error is coming from the `UserProfileAPIView` class `get_serializer_class` method. This happens because we decided not to pass requests object into `SchemaGenerator`  class `get_schema` method because doing so will apply permissions to the schema endpoints and this is not desirable.

One quick fix for this is to replace the `UserProfileAPIView` class `get_serializer_class` method with:

` def get_serializer_class(self):`

`    if self.request and self.kwargs.get("username") != self.request.user.username:`
`        return CreatorMinimalSerializer`
`    else:`
`        return CreatorSerializer`

and add the following to the class docstring:

`Note that this schema returns the full user profile, but the api sometimes` 
`returns a minimal version of the user profile, omitting certain fields that` 
`are not neccessary or are sensitive.`

**To Reproduce**
1. follow the readme on zubhub home page to setup zubhub on your local environment.
2. visit https://localhost:8000 and observe what happens

**Expected behavior**
You should have zubhub API documentation on your screen.

**What Happens**
You get an error page with message 'failed to load api specification'

### PR DESCRIPTION
**Pull request by Somya-Singhal**
Closes #357

## Changes
- modified `UserProfileAPIView` class `get_serializer_class` method.
- modified `UserProfileAPIView` class docstring.

### CODE REVIEW

**Comment by tuxology** [comment]
@Somya-Singhal Thanks :pray: We'll review this shortly!

File: zubhub_backend/zubhub/creators/views.py [hunk 1]
```diff
@@ -91,10 +91,16 @@ class UserProfileAPIView(RetrieveAPIView):
...
     def get_serializer_class(self):
-        if self.kwargs.get("username") == self.request.user.username:
-            return CreatorSerializer
-        else:
+        if self.request and self.kwargs.get("username") != self.request.user.username:
             return CreatorMinimalSerializer
+        else:
+            return CreatorSerializer
+
+    """
```
Comment (NdibeRaymond) [inline zubhub_backend/zubhub/creators/views.py]: The docstring below should be a part of the docstring at the beginning of the class, not here

**Review by NdibeRaymond — CHANGES_REQUESTED** [review]

**Comment by NdibeRaymond** [comment]
hello @Somya-Singhal, thanks for creating this PR! I left a comment on your changes. Once that is fixed, this PR will be merged.

**Comment by NdibeRaymond** [comment]
Also, improvements can be made to your PR description. Try to make the description describe exactly what you did.

**Comment by Somya-Singhal** [comment]
Ok sure @NdibeRaymond, I will make changes shortly and update you.

**Comment by Somya-Singhal** [comment]
Hello @NdibeRaymond @tuxology, I have made the changes as requested. Please do let me know if any further changes are required.

File: zubhub_backend/zubhub/creators/views.py [hunk 2]
```diff
@@ -79,10 +79,9 @@ def get(self, request, format=None):
 
 class UserProfileAPIView(RetrieveAPIView):
     """
-    Fetch Profile of user with given username.
-
-    Requires username of user.
-    Returns user profile.
+    Note that this schema returns the full user profile, but the api sometimes
```
Comment (NdibeRaymond) [inline zubhub_backend/zubhub/creators/views.py:86]: The docstring should be added to the previous docstring, not replace it. It should look more like 
    
"""
Fetch Profile of user with given username.
Requires username of user.
Returns user profile.

Note that this schema returns the full user profile, but the api sometimes
returns a minimal version of the user profile, omitting certain fields that
are not neccessary or are sensitive.
"""

Comment (Somya-Singhal) [reply zubhub_backend/zubhub/creators/views.py:86]: @NdibeRaymond, I have fixed it. Now could you please check?

**Review by NdibeRaymond — APPROVED** [review]

**Comment by NdibeRaymond** [comment]
Great! Something I do most times (and I think makes sense in this case ) is to always rebase multiple commits in a single PR into a single commit before merging, especially for a minor change like this one. Other than that, we are good to merge.

**Comment by Somya-Singhal** [comment]
> Great! Something I do most times (and I think makes sense in this case ) is to always rebase multiple commits in a single PR into a single commit before merging, especially for a minor change like this one. Other than that, we are good to merge.

Sure @NdibeRaymond, I will make sure to do it in my next PR's. Thanks for the information.

**Pull request merged**